src/java/org/apache/cassandra/dht/RangeStreamer.java

```diff
@@ -446,16 +447,14 @@ else if (useStrictConsistency)
                          //The old behavior where we might be asked to fetch ranges we don't need shouldn't occur anymore.
                          //So it's an error if we don't find what we need.
                          if (oldEndpoints.isEmpty() && toFetch.isTransient())
-                         {
                              throw new AssertionError("If there are no endpoints to fetch from then we must be transitioning from transient to full for range " + toFetch);
-                         }
 
-                         if (!any(oldEndpoints, isSufficient))
+                         if (toFetch.isFull() && (oldEndpoints.isEmpty() || oldEndpoints.get(0).isTransient()))
```

Comment:
This doesn't seem clearer to me as a more complex boolean expression.

Reply:
My main problem with `!any` is that it negates. Iteration is harder to reason about when negated in my opinion. But this is fairly minor so I'm fine with leaving it as it was

Reply:
I definitely agree the old expression is clearer.

If you don't like negating `any` (though I find it clearer than `none`, personally, as it reads "it's not the case that any endpoints is/are sufficient" as opposed to "none endpoints is/are sufficient"), I've proposed introducing `any` `all` and `none` in [14726](https://issues.apache.org/jira/browse/CASSANDRA-14726)

Reply:
Ok, reverted to the old one